src/pop/blockchain/alt_block_tree.cpp

```diff
@@ -745,9 +773,11 @@ bool AltBlockTree::finalizeBlockImpl(index_t& index,
   firstBlockHeight = std::max(bootstrapBlockHeight, firstBlockHeight);
   auto* finalizedIndex = vbk().getBestChain()[firstBlockHeight];
   VBK_ASSERT_MSG(finalizedIndex != nullptr, "Invalid VBK tree state");
-  if (!vbk().finalizeBlock(*finalizedIndex, state)) {
-    return state.Invalid("vbktree-finalize-error");
-  }
+  bool result = vbk().finalizeBlock(*finalizedIndex, state);
+  VBK_ASSERT_MSG(result,
+                 "Failed to finalize block %s: %s",
+                 finalizedIndex->toShortPrettyString(),
+                 state.toString());
```

Comment:
It seems that finalizeBlock should never fail. Put this assert inside finalizeBlock() and do not return anything from it.